### PR TITLE
[DNM] ASoC: SOF: Intel: hda-loader: Add support for split librarly loading

### PR DESCRIPTION
**Pull request by ujfalusi**
There is a certain sequence needs to be followed when configuring the HDA DMA in host and DSP.
The firmware provides a way to handle this two stage sequencing by splitting the LOAD_LIBRARY into two stage:
1st stage: the lib_id is 0, used to configure the DMA on DSP side 2nd stage: both dma_id and lib_id is valid, used for the actual transfer of the library

In case a firmware without support for this two stage loading is used then the second stage message will trigger the loading and the first stage will return with error, which is ignored by the kernel.

### CODE REVIEW

**Comment by ujfalusi** [comment]
This PR is meant to be used along with https://github.com/thesofproject/sof/pull/7986 which implements the two stage LIBRARY_LOADING.
Without that PR the SOF firmware will lock up due to missing lib_id validation.

**Comment by plbossart** [comment]
> This PR is meant to be used along with [thesofproject/sof#7986](https://github.com/thesofproject/sof/pull/7986) which implements the two stage LIBRARY_LOADING.
> Without that PR the SOF firmware will lock up due to missing lib_id validation.

that seems to contradict the previous statement "In case a firmware without support for this two stage loading is used then the second stage message will trigger the loading and the first stage will return with error, which is ignored by the kernel."

Not getting the picture at all @ujfalusi

File: sound/soc/sof/intel/hda-loader.c [hunk 1]
```diff
@@ -549,14 +549,26 @@ int hda_dsp_ipc4_load_library(struct snd_sof_dev *sdev,
...
+	/*
+	 * LOAD_LIBRARY 1st stage
+	 * message with valid dma_id and invalid (0) lib_id
+	 * Firmware can respond with error if it is not prepared for the two
+	 * stage library loading, the second stage message will in both case
+	 * going to complete the loading.
+	 * Error from sof_ipc_tx_message_no_reply() is ignored at this stage.
+	 */
+	msg.primary |= SOF_IPC4_GLB_LOAD_LIBRARY_LIB_ID(0);
+	sof_ipc_tx_message_no_reply(sdev->ipc, &msg, 0);
```
Comment (ranj063) [inline sound/soc/sof/intel/hda-loader.c:562]: @ujfalusi i understand we want to ignore the error for the test firmware which doesnt need 2-state loading but ignoring the error in the case of SOF firmware will be bad. I think we should drop the support for the tester in our driver

**Comment by ujfalusi** [comment]
Closing this PR as the splitting of LOAD_LIBRARY is a wrong way to handle this.

https://github.com/thesofproject/linux/pull/4496 is the new proposal to honor the hardware programming sequence.

**Pull request closed without merging**